pandas/core/frame.py

```diff
@@ -12004,6 +12010,25 @@ def c(x):
 
         return correl
 
+    def _transform_ord_cat_cols_to_coded_cols(self) -> DataFrame:
+        """
+        any ordered categorical columns are transformed to the respective
+        categorical codes while other columns remain untouched
+        """
+        categ = self.select_dtypes("category")
+        if len(categ.columns) == 0:
+            return self
+
+        cols_convert = categ.loc[:, categ.agg(lambda x: x.cat.ordered)].columns
+
+        if len(cols_convert) > 0:
+            data = self.copy(deep=False)
```

Comment:
I'm a bit wary of taking an entire copy of the dataframe in instances where there might be ordered categoricals; that's a potentially large performance hit, and the usage of this seems pretty niche

I see @rhshadrach commented on the original issue, so lets see what his thoughts are

Reply:
`deep=False` shouldn't be large as it doesn't copy the underlying data, but agreed we should measure the performance here.

Reply:
For benchmarking, we don't have any ASVs that hit this case. You can just setup an example that hits this case and use `timeit` to compare this PR to main. Aim for 10-100ms in runtime so we aren't merely benchmarking overhead. If you want any assistance in setting this up, just let me know.